packages/docusaurus-theme-search/src/theme/SearchBar/index.tsx

```diff
@@ -0,0 +1,170 @@
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import { MarkpromptConfig } from '@markprompt/docusaurus-theme-search';
+import * as Markprompt from '@markprompt/react';
+import { VisuallyHidden } from '@radix-ui/react-visually-hidden';
+import React, { type ReactElement, useContext } from 'react';
+
+import styles from './index.module.css';
+
+export default function SearchBar(): ReactElement {
+  const { siteConfig } = useDocusaurusContext();
+  const markpromptConfig = siteConfig.themeConfig
+    .markprompt as MarkpromptConfig;
+
+  return (
+    <Markprompt.Root {...markpromptConfig}>
+      <Markprompt.Trigger
+        aria-label="Open Markprompt"
+        className={styles.MarkpromptButton}
+      >
+        <ChatIcon className={styles.MarkpromptIcon} />
+      </Markprompt.Trigger>
+      <Markprompt.Portal>
+        <Markprompt.Overlay className={styles.MarkpromptOverlay} />
+        <Markprompt.Content className={styles.MarkpromptContent}>
+          <Markprompt.Close className={styles.MarkpromptClose}>
+            <CloseIcon />
+          </Markprompt.Close>
+
+          {/* Markprompt.Title is required for accessibility reasons. It can be hidden using an accessible content hiding technique. */}
+          <VisuallyHidden asChild>
+            <Markprompt.Title>
+              Ask me anything about Markprompt
```

Comment:
How would folks add their own title and description? Through the config as well?

Reply:
Docusaurus has a concept called [_Swizzling_](https://docusaurus.io/docs/swizzling). This basically means a component (`SearchBar` source directory) gets copied into their project, and they can modify everything.